lib/_tls_common.js

```diff
@@ -193,113 +244,113 @@ exports.createSecureContext = function createSecureContext(options) {
-                                    _.length > 0).join(':');
-  const cipherSuites = ciphers.filter((_) => _.match(/^TLS_/)).join(':');
-
-  if (cipherSuites === '' && cipherList === '') {
-    // Specifying empty cipher suites for both TLS1.2 and TLS1.3 is invalid, its
-    // not possible to handshake with no suites.
-    throw new ERR_INVALID_ARG_VALUE('options.ciphers', ciphers);
-  }
+  const { cipherList, cipherSuites } = processCiphers(ciphers);
 
   c.context.setCipherSuites(cipherSuites);
   c.context.setCiphers(cipherList);
 
-  if (cipherSuites === '' && c.context.getMaxProto() > TLS1_2_VERSION &&
-      c.context.getMinProto() < TLS1_3_VERSION)
+  if (cipherSuites === '' &&
+      c.context.getMaxProto() > TLS1_2_VERSION &&
+      c.context.getMinProto() < TLS1_3_VERSION) {
     c.context.setMaxProto(TLS1_2_VERSION);
+  }
 
-  if (cipherList === '' && c.context.getMinProto() < TLS1_3_VERSION &&
-      c.context.getMaxProto() > TLS1_2_VERSION)
+  if (cipherList === '' &&
+      c.context.getMinProto() < TLS1_3_VERSION &&
+      c.context.getMaxProto() > TLS1_2_VERSION) {
     c.context.setMinProto(TLS1_3_VERSION);
+  }
 
-  if (options.ecdhCurve === undefined)
-    c.context.setECDHCurve(tls.DEFAULT_ECDH_CURVE);
-  else if (options.ecdhCurve)
-    c.context.setECDHCurve(options.ecdhCurve);
+  validateString(ecdhCurve, 'options.ecdhCurve');
+  c.context.setECDHCurve(ecdhCurve);
 
-  if (options.dhparam) {
-    const warning = c.context.setDHParam(options.dhparam);
+  if (dhparam !== undefined) {
+    validateKeyOrCertOption('dhparam', dhparam);
+    const warning = c.context.setDHParam(dhparam);
     if (warning)
       process.emitWarning(warning, 'SecurityWarning');
   }
 
-  if (options.crl) {
-    if (ArrayIsArray(options.crl)) {
-      for (const crl of options.crl) {
-        c.context.addCRL(crl);
+  if (crl !== undefined) {
+    if (ArrayIsArray(crl)) {
+      for (const val of crl) {
+        validateKeyOrCertOption('crl', val);
+        c.context.addCRL(val);
       }
     } else {
-      c.context.addCRL(options.crl);
+      validateKeyOrCertOption('crl', crl);
+      c.context.addCRL(crl);
     }
   }
 
-  if (options.sessionIdContext) {
-    c.context.setSessionIdContext(options.sessionIdContext);
+  if (sessionIdContext !== undefined) {
+    validateString(sessionIdContext, 'options.sessionIdContext');
+    c.context.setSessionIdContext(sessionIdContext);
   }
 
-  if (options.pfx) {
-    if (!toBuf)
-      toBuf = require('internal/crypto/util').toBuf;
-
-    if (ArrayIsArray(options.pfx)) {
-      for (const pfx of options.pfx) {
-        const raw = pfx.buf ? pfx.buf : pfx;
-        const buf = toBuf(raw);
-        const passphrase = pfx.passphrase || options.passphrase;
-        if (passphrase) {
-          c.context.loadPKCS12(buf, toBuf(passphrase));
+  if (pfx !== undefined) {
+    if (ArrayIsArray(pfx)) {
+      for (const val of pfx) {
+        const raw = val.buf ? val.buf : val;
```

Comment:
nit
```suggestion
        const raw = val.buf ?? val;
```

Reply:
For now I'm going to skip these until I can do some more performance impact profiling on these new operators in general